inginious/frontend/user_manager.py

```diff
@@ -512,14 +512,24 @@ def revoke_binding(self, username, binding_id):
             msg = _("You must set a password before removing all bindings.")
         return error, msg
 
-    def delete_user(self, username):
+    def delete_user(self, username, confirmation_email=None):
         """
-        Delete a user
+        Delete a user based on username
         :param username: the username of the user
+        :param confirmation_email: An email to confirm suppression. May be None
         :return a boolean if a user was deleted
         """
-        result = self._database.users.delete_one({"username": username})
-        return result.deleted_count == 1
+        query = {"username": username, "email": confirmation_email} \
+            if confirmation_email is not None else {"username": username}
+        result = self._database.users.find_one_and_delete(query)
+        if not result:
+            return False
+        else:
+            self._database.submissions.delete_many({"username": username})
+            self._database.user_tasks.delete_many({"username": username})
+            all_courses = self._database.courses.find({})
```

Comment:
It would be better to only select the user's courses, even if the ``unregister_user`` method can be run with non-registered users, to avoid additional useless requests ans false logging messages (User xxx unregistered from...)

Reply:
I'm intent to resolve this like that:
`user_courses = [course['_id'] for course in self._database.courses.find({})
                            if username in course["students"]]`
Seems good to you?

Reply:
You can simply use the username as filter. ``{"students": username}``